Fix .success() method in Results

**Description**
The Results.success() method returns true if there are successes, but does not check that there are no failures

It should return true if successes >1 and failures == 0.

Make sure the tests are passing, and if not please discuss the test failures with us.

hey @joelrosario  im intrested contributing in this issue ,and i've also get solution but for me to test the code ,when i try to  ./gradlew clean build
my some test cases are getting failed



```plaintext
Task :application:test
Java HotSpot(TM) 64-Bit Server VM warning: Sharing is only supported for boot lo
BundleCommandTest > the command should compress all stub commands in the config into a zip file() FAILED

BundleCommandTest > should list stub files in subdirectories of the given directory() FAILED

BundleCommandTest > should list stub files in a given directory() FAILED
org.opentest4j.AssertionFailedError at BundleCommandTest.kt:72

BundleCommandTest > should avoid entries that are neither file nor directory() FAILED
java.lang.NullPointerException at BundleCommandTest.kt:120

BundleCommandTestE2E > basic stub bundle command functionality(File) FAILED
java.nio.file.InvalidPathException at BundleCommandTestE2E.kt:131

BundleCommandTestE2E > bundle is generated with shifted base(File) FAILED
java.nio.file.InvalidPathException at BundleCommandTestE2E.kt:164

BundleCommandTestE2E > test bundle is generated(File) FAILED
java.nio.file.InvalidPathException at BundleCommandTestE2E.kt:208

CentralContractRepoReportCommandTestE2E > test generates report json file() FAILED

StubCommandTest > when a contract with the incorrect extension command should exit with non-zero(Path) FAILED

66 tests completed, 9 failed, 1 skipped

Task :application:test FAILED

FAILURE: Build failed with an exception.

What went wrong:
Execution failed for task ':application:test'.
There were failing tests. See the report at: file:///C:/Users/Tejas/Desktop/new%20open/specmatic/application/build/reports/tests/test/index.html

Try:
Run with --stacktrace option to get the stack trace.
Run with --info or --debug option to get more log output.
Run with --scan to get full insights.

Get more help at https://help.gradle.org/
BUILD FAILED in 1m 15s
30 actionable tasks: 30 executed
```
![Screenshot (127)](https://github.com/znsio/specmatic/assets/98216813/a1bcba8a-bca4-4de3-bfa4-ef17c3f93549)
![Screenshot (128)](https://github.com/znsio/specmatic/assets/98216813/d45f01d4-7c10-478a-b40f-b848e65936bc)
![Screenshot (129)](https://github.com/znsio/specmatic/assets/98216813/564d433e-ae0f-4e86-baea-38187692740e)
![Screenshot (130)](https://github.com/znsio/specmatic/assets/98216813/50853569-e8fb-4691-aee9-4c96b23f8797)

--- a/application/src/main/kotlin/application/CompatibleCommand.kt
+++ b/application/src/main/kotlin/application/CompatibleCommand.kt
@@ -360,8 +360,8 @@ internal data class CompatibilityOutput(val exitCode: Int, val message: String)
 internal fun compatibilityMessage(results: Outcome<Results>): CompatibilityOutput {
     return when {
         results.result == null -> CompatibilityOutput(1, results.errorMessage)
-        results.result.success() -> CompatibilityOutput(0, results.errorMessage.ifEmpty { "The newer contract is backward compatible" })
-        else -> CompatibilityOutput(1, compatibilityReport(results.result, "The newer contract is NOT backward compatible"))
+        results.result.hasFailures() -> CompatibilityOutput(1, compatibilityReport(results.result, "The newer contract is NOT backward compatible"))
+        else -> CompatibilityOutput(0, results.errorMessage.ifEmpty { "The newer contract is backward compatible" })
     }
 }
 

--- a/core/src/main/kotlin/in/specmatic/core/Results.kt
+++ b/core/src/main/kotlin/in/specmatic/core/Results.kt
@@ -6,7 +6,7 @@ data class Results(val results: List<Result> = emptyList()) {
     fun hasResults(): Boolean = results.isNotEmpty()
 
     fun hasFailures(): Boolean = results.any { it is Result.Failure }
-    fun success(): Boolean = !hasFailures()
+    fun success(): Boolean = successCount > 0 && failureCount == 0
 
     fun withoutFluff(fluffLevel: Int): Results = copy(results = results.filterNot { it.isFluffy(fluffLevel) })
 

--- a/core/src/main/kotlin/in/specmatic/core/Results.kt
+++ b/core/src/main/kotlin/in/specmatic/core/Results.kt
@@ -6,7 +6,7 @@ data class Results(val results: List<Result> = emptyList()) {
     fun hasResults(): Boolean = results.isNotEmpty()
 
     fun hasFailures(): Boolean = results.any { it is Result.Failure }
-    fun success(): Boolean = successCount > 0 && failureCount == 0
+    fun success(): Boolean = if(hasResults()) successCount > 0 && failureCount == 0 else true
 
     fun withoutFluff(fluffLevel: Int): Results = copy(results = results.filterNot { it.isFluffy(fluffLevel) })
 

--- a/core/src/test/kotlin/in/specmatic/core/TestBackwardCompatibilityKtTest.kt
+++ b/core/src/test/kotlin/in/specmatic/core/TestBackwardCompatibilityKtTest.kt
@@ -884,7 +884,7 @@ Then status 200
 
         assertThat(results.successCount).isZero()
         assertThat(results.failureCount).isZero()
-        assertThat(results.success()).isTrue()
+        assertThat(results.hasFailures()).isFalse()
     }
 
     @Test

